File: Python/POO I/Exercicios 22/Forno.py
# -*- coding: utf-8 -*-
class Forno:
    __temp = 50
    __status = 0
    def __init__(self,marca, tempm):
        self.__marca = marca
        self.__tempm = tempm

    def setLigar(self):
        self.__status = 1
    def setDesligado(self):
        self.__status = 0
    def setAumentar(self,aumentar):
        if self.__temp == self.__tempm:
            return False
        elif (self.__temp + aumentar) > self.__tempm:
            return False
        else:
            self.__temp += aumentar
    def setDiminuir(self,diminuir):
        if self.__temp == 50:
            return False
        elif (self.__temp - diminuir) < 50:
            return False
        else:
            self.__temp -= diminuir

    def getTemp(self):
        return self.__temp
    def getStatus(self):
        return self.__status
    def getMarca(self):
        return self.__marca

#A
t = Forno('Tabajra',100)
#B
o = Forno('Orange',350)
#C
t.setLigar()
#D
print(t.getTemp())
#E
t.setAumentar(25)
#F
print(t.getTemp())
#G
t.setAumentar(30)
#H
print(t.getTemp())
#I
t.setDiminuir(15)
#J
print(t.getTemp())
#K
t.setDiminuir(75)

print()

#L
o.setLigar()
#M
print(o.getTemp())
#N
o.setAumentar(300)
#O
print(o.getTemp())
#P
o.setDiminuir(20)
#Q
print(o.getTemp())
#R
o.setAumentar(30)
#S
print(o.getTemp())

print()

#T
print(o.getStatus())
#U
print(t.getStatus())

#V
o.setDesligado()
t.setDesligado()
#W
print(o.getStatus())
#X
print(t.getStatus())

print()

#Y
print(t.getMarca())
print(o.getMarca())

File: C++/Forno.cpp
#include "Forno.hpp"
#include <vector>

int main(){
    vector<Forno>lista;

    Forno tabajara("Tabajara",100);
    Forno orange("Orange",350);

    tabajara.ligar();
    printf("%d\n",tabajara.getTemp());
    tabajara.mudarTemperatura(25);
    printf("%d\n",tabajara.getTemp());
    tabajara.mudarTemperatura(30);
    printf("%d\n",tabajara.getTemp());
    tabajara.mudarTemperatura(-15);
    printf("%d\n",tabajara.getTemp());
    tabajara.mudarTemperatura(-75);

    orange.ligar();
    printf("%d\n",orange.getTemp());
    orange.mudarTemperatura(300);
    printf("%d\n",orange.getTemp());
    orange.mudarTemperatura(-20);
    printf("%d\n",orange.getTemp());
    orange.mudarTemperatura(30);
    printf("%d\n",orange.getTemp());
    printf("%d\n",orange.getStatus());
    printf("%d\n",tabajara.getStatus());

    tabajara.desligar();
    orange.desligar();

    printf("%d\n",orange.getStatus());
    printf("%d\n",tabajara.getStatus());

    lista.push_back(tabajara);
    lista.push_back(orange);

    for (auto & value:lista){
        printf("Marca: %s\nTemperatura: %d\n",value.getMarca().c_str(),value.getTemp());
    }






    return 0;
}